source/Halibut/Transport/Protocol/ITypeRegistry.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
+namespace Halibut.Transport.Protocol
+{
+    public interface ITypeRegistry
```

Comment:
The name `ITypeRegistry` feels like it has more purpose beyond serialisation. Does it have a future beyond serialization?

If not, maybe a different name might suffice? `ISeriazableTypeRegistry`?